src/main/java/org/folio/service/pieces/PieceStorageService.java

```diff
@@ -134,21 +137,19 @@ public Future<PieceCollection> getAllPieces(int limit, int offset, String query,
     return restClient.get(requestEntry, PieceCollection.class, requestContext);
   }
 
-  private Future<List<Piece>> filterPiecesByUserTenantsIfNecessary(List<Piece> pieces, RequestContext requestContext) {
+  private Future<List<String>> getUserTenantsIfNeeded(RequestContext requestContext) {
     return consortiumConfigurationService.getConsortiumConfiguration(requestContext)
-      .compose(consortiumConfiguration -> consortiumConfiguration
-        .filter(configuration -> shouldFilterPiecesForTenant(configuration.centralTenantId(), requestContext))
-        .map(configuration -> consortiumUserTenantsRetriever.getUserTenants(configuration.consortiumId(), requestContext)
-          .map(userTenants -> filterPiecesByUserTenants(pieces, userTenants)))
-        .orElse(Future.succeededFuture(pieces)));
-  }
-
-  private List<Piece> filterPiecesByUserTenants(List<Piece> pieces, List<String> userTenants) {
-    return pieces.stream()
-      .filter(piece -> Optional.ofNullable(piece.getReceivingTenantId())
-        .map(userTenants::contains)
-        .orElse(true))
-      .toList();
+      .compose(consortiumConfiguration -> {
+        if (consortiumConfiguration.isEmpty()) {
+          return Future.succeededFuture();
+        }
+        var configuration = consortiumConfiguration.get();
+        return settingsRetriever.getSettingByKey(SettingKey.CENTRAL_ORDERING_ENABLED, requestContext)
```

Comment:
Module descriptors of endpoints that use getPieces should be updated with the new module permission